osquery/core/system.cpp

```diff
@@ -188,23 +194,37 @@ Status getHostUUID(std::string& ident) {
   return status;
 }
 
+Status getSpecifiedUUID(std::string& ident) {
+  if (FLAGS_specified_identifier.empty()) {
+    return Status(1, "No specified identifier for host");
+  }
+  ident = FLAGS_specified_identifier;
+  return Status(0, "OK");
+}
+
 std::string getHostIdentifier() {
   static std::string ident;
 
+  Status result;
   if (ident.size() == 0) {
     if (FLAGS_host_identifier == "uuid") {
-      getHostUUID(ident);
+      result = getHostUUID(ident);
     } else if (FLAGS_host_identifier == "instance") {
-      getInstanceUUID(ident);
+      result = getInstanceUUID(ident);
     } else if (FLAGS_host_identifier == "ephemeral") {
-      getEphemeralUUID(ident);
+      result = getEphemeralUUID(ident);
+    } else if (FLAGS_host_identifier == "specified") {
+      result = getSpecifiedUUID(ident);
     } else {
       // assuming the default of "hostname" as the machine identifier
       // intentionally not set to `ident` because the hostname may change
       // throughout the life of the process and we always want to be using the
       // most current hostname
       return osquery::getHostname();
     }
+    if (!result.ok()) {
```

Comment:
I'm not sure I follow. Do you mean to move the above `return osquery::getHostname()` into the `if (!result.ok()) {` case? Or are you trying to differentiate when `FLAGS_host_identifier` is not one of the expected values from when the get*UUID function fails?